cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -25,18 +25,22 @@ import (
 )
 
 type thresholdBasedEstimationLimiter struct {
-	maxDuration time.Duration
-	maxNodes    int
-	nodes       int
-	start       time.Time
+	maxDuration     time.Duration
+	maxNodes        int
+	maxNodesDefault int
+	nodes           int
+	start           time.Time
 }
 
-func (tbel *thresholdBasedEstimationLimiter) StartEstimation([]*apiv1.Pod, cloudprovider.NodeGroup) {
+func (tbel *thresholdBasedEstimationLimiter) StartEstimation(_ []*apiv1.Pod, _ cloudprovider.NodeGroup, runtimeLimits []BinpackingLimit) {
 	tbel.start = time.Now()
 	tbel.nodes = 0
+	tbel.maxNodes = computeLimit(runtimeLimits, tbel.maxNodes)
```

Comment:
This also relies on the limits passed during initialisation returning constant values - we should also include them here. Plus this way we won't need the second argument in the `computeLimit` method.

Reply:
initializing maxNodes & maxDuration was refactored - 2742946b9ca375b584f7a2ef2eedd69b8a736fea